aws/ec2.go

```diff
@@ -0,0 +1,37 @@
+package aws
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/client"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
+)
+
+//EC2 is an ec2 svc
+type EC2 struct {
+	Svc ec2iface.EC2API
+}
+
+// NewEC2 returns a new EC2 svc
+func NewEC2(c client.ConfigProvider, config *aws.Config) *EC2 {
+	return &EC2{Svc: ec2.New(c, config)}
+}
+
+// GetAllInstances will walk all instances and call func for each
+func (e *EC2) GetAllInstances(f func(*ec2.Instance)) error {
+	var err error
+	input := &ec2.DescribeInstancesInput{}
+	err = e.Svc.DescribeInstancesPages(input, func(output *ec2.DescribeInstancesOutput, lastPage bool) bool {
+		for _, reservation := range output.Reservations {
+			if reservation == nil {
+				continue
+			}
+			for _, instance := range reservation.Instances {
+				f(instance)
+			}
+		}
+		return true
+	})
+
+	return err
```

Comment:
i've been trying to errors.Wrap() these so we get the stack "at the source"

Reply:
yeah sure